Retarus/retarus_modules/retarus_events_comsumer.py

```diff
@@ -0,0 +1,63 @@
+from queue import SimpleQueue
+from threading import Event, Thread
+import time
+
+import orjson
+from sekoia_automation.connector import Connector
+from websocket import WebSocketApp, WebSocketTimeoutException
+
+
+class RetarusEventsConsumer(Thread):
+    def __init__(self, connector: Connector, queue: SimpleQueue):
+        super().__init__()
+        self.connector = connector
+        self.queue = queue
+        self.configuration = connector.configuration
+        self._stop_event = Event()
+
+    def create_websocket(self):
+        websocket = WebSocketApp(
+            url=self.configuration.ws_url,
+            header=[f"Authorization: Bearer {self.configuration.ws_key}"],
+            on_message=self.on_message,
+            on_error=self.on_error,
+            on_close=self.on_close,
+        )
+        websocket_thread = Thread(target=websocket.run_forever)
+        websocket_thread.daemon = True
+        websocket_thread.start()
+
+        return websocket, websocket_thread
+
+    def on_message(self, _, event) -> None:
+        try:
+            message = orjson.loads(event)
+            self.queue.put(message)
+        except orjson.JSONDecodeError as ex:
+            self.connector.log_exception(ex, message="Failed to consume event")
+
+    def on_error(self, _, error):
+        if isinstance(error, WebSocketTimeoutException):
+            # add a gentler message for timeout
+            self.connector.log(message="Websocket timed out", level="warning")
+        else:
+            self.connector.log(message=f"Websocket error: {error}", level="error")
+
+    def on_close(self, *_):
+        self.connector.log(message="Closing socket connection", level="info")
+
+    @property
+    def is_running(self):
+        return not self._stop_event.is_set()
+
+    def run(self):
+        self.connector.log(message=f"Connection to stream {self.configuration.ws_url}", level="info")
+
+        websocket, websocket_thread = self.create_websocket()
+        websocket_thread.start()
+
+        while self.is_running:
+            time.sleep(1)
+            pass
```

Comment:
You can wait until the flag is set

```suggestion
        self._stop_event.wait()
```

Reply:
Didn't now, awesome